src/libcore/ops/range.rs

```diff
@@ -854,14 +854,13 @@ impl<'a, T: ?Sized + 'a> RangeBounds<T> for (Bound<&'a T>, Bound<&'a T>) {
     fn start_bound(&self) -> Bound<&T> {
         self.0
     }
-
     fn end_bound(&self) -> Bound<&T> {
         self.1
     }
 }
 
 #[stable(feature = "collections_range", since = "1.28.0")]
-impl<T> RangeBounds<T> for RangeFrom<&T> {
+impl<'a, T: ?Sized + 'a> RangeBounds<T> for RangeFrom<&'a T> {
```

Comment:
The `?Sized` bounds here and below look good. (Although adding an explicit `'a` lifetime parameter is unnecessary.) The changes above however likely should be reverted. (See my other comment.)